drivers/display/ls0xx.c

```diff
@@ -0,0 +1,493 @@
+LOG_MODULE_REGISTER(ls0xx);
+
+#include <string.h>
+#include <device.h>
+#include <drivers/display.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+#include <display/cfb.h>
+
+/* Note:
+ * -> .write API should be called atleast once per second so that VCOM can 
+ *      be toggled. If there is nothing to update then buf can be set to NULL
+ *      or, width/height can be set to 0 so the driver will only 
+ *      toggle VCOM and return
+ * -> high/1 means white, low/0 means black
+ * -> Display expects LSB first
+ * -> If devicetree contains DISP pin then it will be set to high
+ *      during initialization.
+ * -> VCOM is toggled through software by default (CONFIG_LS0XX_EXTMODE_SERIAL=y)
+ *    EXTMODE pin is connected to VSS 
+ * -> To control VCOM using external signal EXTCOMIN, set
+ *      CONFIG_LS0XX_EXTMODE_SERIAL=n
+ *      EXTMODE pin is connected to VDD
+ * -> When designing with a memory LCD panel, a decision must be made as to how
+ *      VCOM will be generated. VCOM is an alternating signal that prevents a DC bias
+ *      from being built up within the panel. If this DC bias is allowed to
+ *      accumulate, it will eventually reach the point where the state of the
+ *      liquid crystals in the panel cells cannot be changed by the panel electronics
+ *      and the cells will no longer change state. Memory LCDs do not generate
+ *      this signal internally. It must be supplied using one of two methods:
+ *      software, or external clock. The mode is selected by the EXTMODE pin on
+ *      the interface. If external clock is selected [EXTMODE = H], the clock
+ *      should be supplied on the EXTCOMM pin.
+ *      When the software clock is selected [EXTMODE = L], bit V of the command
+ *      bit string sets the state of VCOM. This bit must toggle
+ *      (by writing to the panel) at least once per second. Any command string
+ *      can be used to change the state of the VCOM bit. If no update to the
+ *      pixel memory is needed within a second, the ‘Change VCOM’ command can be
+ *      used to toggle the bit state. In the descriptions and figures,
+ *      the V-bit represents the desired VCOM state when software VCOM control
+ *      is selected. When the external clock is selected, then the V-bit state
+ *      doesn’t matter. To toggle VCOM/EXTCOMIN use ls0xx_toggle_extcomin from
+ *      application code as described above.
+ *      Source:
+ * https://www.sharpsde.com/fileadmin/products/Displays/2016_SDE_App_Note_for_Memory_LCD_programming_V1.3.pdf
+ * -> Frame buf format for each row, This format is for LS0XX_PANEL_HEIGHT
+ *      Adding 2 to frame_buf for the line number and dummy byte
+ * +-------------------+-------------------+----------------+
+ * | line num (8 bits) | data (WIDTH bits) | dummy (8 bits) |
+ * +-------------------+-------------------+----------------+
+ */
+
+#define LS0XX_SPI_FREQ            DT_INST_PROP(0, spi_max_frequency)
```

Comment:
This define, how it is used now, seems not to be necessary, you can use DT_INST_PROP(0, spi_max_frequency) directly in the L428. Please check the others as well.